### PR TITLE
Added Featurizers/CMakeLists.txt

### PR DESCRIPTION
**Pull request by davidbrownellWork**

### CODE REVIEW

File: src/Featurizers/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# ----------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License
+# ----------------------------------------------------------------------
+cmake_minimum_required(VERSION 3.5.0)
+
+macro(Impl)
+    # Defining a function here to introduce a new scope for local variables
+    get_filename_component(_this_path ${CMAKE_CURRENT_LIST_FILE} DIRECTORY)
```
Comment (michaelgsharp) [inline src/Featurizers/CMakeLists.txt]: I'm just curious as I don't know cmake well enough, but where does the `${CMAKE_CURRENT_LIST_FILE}` come from?

File: src/Featurizers/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    include(${_this_path}/cmake/FeaturizersCode.cmake)
```
Comment (michaelgsharp) [inline src/Featurizers/CMakeLists.txt]: On line 9 you have `_this_path`, but on line 11 you have `${_this_path}`, was that intended?

File: src/Featurizers/CMakeLists.txt (hunk 1 above)
Comment (davidbrownellWork) [reply src/Featurizers/CMakeLists.txt]: [cmake provides it](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_LIST_FILE.html).

File: src/Featurizers/CMakeLists.txt (hunk 2 above)
Comment (yuslepukhin) [inline src/Featurizers/CMakeLists.txt]: >${_this_path}/ [](start = 12, length = 14)

Can we remove ${_this_path}/  and just do cmake/Featurizers.code ? Anyone can read it.

Comment (davidbrownellWork) [reply src/Featurizers/CMakeLists.txt]: cmake syntax is awful - the usage is correct.

Comment (yuslepukhin) [reply src/Featurizers/CMakeLists.txt]: Let's make it readable

---
In reply to: [356252729](https://github.com/microsoft/FeaturizersLibrary/pull/25#discussion_r356252729) [](ancestors = 356252729)

**Pull request merged**